public/app/plugins/datasource/tempo/resultTransformer.ts

```diff
@@ -282,7 +289,8 @@ export function transformFromOTLP(
             serviceTags,
             startTime: span.startTimeUnixNano! / 1000000,
             duration: (span.endTimeUnixNano! - span.startTimeUnixNano!) / 1000000,
-            tags: getSpanTags(span, librarySpan.instrumentationLibrary),
+            tags: getSpanTags(span),
+            intrinsics: getIntrinsics(span, librarySpan.instrumentationLibrary),
```

Comment:
I feel like this is a part I am having a different view of what intrinsics are so want to align or that. What I thought intrinsic means is just intrinsic attributes of the trace or span, not a different set of tags. So for example here `duration` is also intrinsic attribute because is it part of the span type. So this to me right now feel a bit odd in the same sense as if `startTime` and `duration` were part of some `intrinsics` map with `duration` key and value instead of just part of the actual type. Then when presenting and we have to have a separate list of keys, that we have to pull out of here first and then display it feels weird but at the same time reason I suggested to put it here because I thought they are just part of the type.

Reply:
When you describe `duration` & `startTime` like that they do sound to be something that could go in the `intrinsics` map, however `duration` & `startTime` always exist on the span whereas not every span has the library intrinsics. Feels a bit tidier to put them in their own map in this sense. I wanted to keep them out of the tags map so we didn't have to filter when adding the tags to the UI however we could just put all of these on the span type and check if they exist before rendering in span details where they are now (next to duration, child count etc)

So to be super clear we would add the following to the span type;
```
kind: enum,
status: {
   code: number,
   message: string
},
instrumentationLibrary: {
   name: string,
   version: string 
}
```
btw have you ever seen the [trace state](https://github.com/grafana/grafana/pull/69394/files#diff-0fae7cc1f8c2a5f78b973620a1894c4a8e848581cc65e3ac2fc603b6a37fb64fL213) in a span?

Reply:
> duration & startTime always exist on the span whereas not every span has the library intrinsics.

Ah ok yeah this is probably what I wasn't clear about. But in what cases we don't have them? I would assume the library and lib-version may be only from OTEL spans but kind and status are sort of universal, aren't they?

Also it's not like we show anything as intrinsic right? Like we have a hardcoded list of things that go there so it's not dynamic like somebody could send a span with some other intrinsics right?

Reply:
Yes, they are so if they are set on the span they will be rendered.

That's it yeah, we have a hardcoded list of intrinsics that may vary in length depending on which exist on the span.